streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractTask.java

```diff
@@ -61,28 +64,42 @@
                  final ChangelogReader changelogReader,
                  final boolean isStandby,
                  final StateDirectory stateDirectory,
-                 final ThreadCache cache) {
+                 final ThreadCache cache,
+                 final StreamsConfig config) {
         this.id = id;
         this.applicationId = applicationId;
         this.partitions = new HashSet<>(partitions);
         this.topology = topology;
         this.consumer = consumer;
         this.cache = cache;
+        eosEnabled = StreamsConfig.EXACTLY_ONCE.equals(config.getString(StreamsConfig.PROCESSING_GUARANTEE_CONFIG));
 
         logPrefix = String.format("%s [%s]", isStandby ? "standby-task" : "task", id());
 
         // create the processor state manager
         try {
-            stateMgr = new ProcessorStateManager(id, partitions, isStandby, stateDirectory, topology.storeToChangelogTopic(), changelogReader);
+            stateMgr = new ProcessorStateManager(
+                id,
+                partitions,
+                isStandby,
+                stateDirectory,
+                topology.storeToChangelogTopic(),
+                changelogReader,
+                eosEnabled);
         } catch (final IOException e) {
             throw new ProcessorStateException(String.format("%s Error while creating the state manager", logPrefix), e);
         }
     }
 
     public abstract void resume();
-    public abstract void commit();
+
+    /**
+     * @param startNewTransaction {@link Producer#beginTransaction() start a new transaction} after successful commit
```

Comment:
It's no public API change -- it's package `internal` -- on need to cover this in a KIP.